vignettes/customization.Rmd

```diff
@@ -0,0 +1,222 @@
+---
+title: "Customize your pkgdown website"
+output: rmarkdown::html_vignette
+description: >
+  Learn how to change the look and feel of pkgdown websites.
+vignette: >
+  %\VignetteIndexEntry{customization}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+```{r setup}
+library(pkgdown)
+```
+
+You can alter the design/styling of your pkgdown website to suit your brand and needs. 
+_Theming_ refers to being able to tweak colors, fonts, spacing etc. by using a Bootswatch template, Bootstrap variables, Bootstrap rules, or any custom CSS. 
+_Layout_ refers to being able to change the content of say the navbar, authors page, `<head>` tag by using pkgdown configuration or pkgdown templates.
+
+Another key concept related to styling your pkgdown website is whether you are tweaking _one_ package website, or several ones. 
+In the latter case you might want to share your style files in the form of a package.
+
+In this vignette we shall explain you the scope and tools for _theming_ and tweaking the _layout_ of pkgdown websites, and then explain how to best distribute your style when applied to more than one pkgdown website.
+
+## Pre-requisite: update to Bootstrap 4!
+
+Note that most theming features work only with Bootstrap 4, and that current development of pkgdown templates only happen for Bootstrap 4 templates. 
+If you are reading this vignette, please take a moment to update your website to Boostrap 4.
+
+```yaml
+template:
+  bootstrap: 4
+```
+
+It should not break your website, it should however change its styling a bit, for the better we hope.
+However, if you were managing your [own pkgdown templates](#layout) or custom CSS, you might need to tweak those.
+
+## Theming
+
+### Choosing a bootswatch theme
+
+You can choose any [Bootswatch theme](https://bootswatch.com/) that's compatible with the Bootstrap version you use.
+Run either `bslib::bootswatch_themes(3)` or `bslib::bootswatch_themes(4)` to get a list.
+If you were using the paper theme, note that it was renamed [materia](https://blog.bootswatch.com/post/167233472837/announcing-bootswatch-400).
+
+To choose a Bootswatch theme here's what to write in your `pkgdown` configuration.
+
+```yaml
+template:
+  bootstrap: 4
+  params:
+    bootswatch: materia
+```
+
+_Some Bootswatch templates like lux or pulse won't work out of the box because of their tall navbar._
+_Add [CSS rules to prevent overlap with other elements](https://getbootstrap.com/docs/4.6/components/navbar/#placement) like [padding to the body](https://stackoverflow.com/questions/10336194/top-nav-bar-blocking-top-content-of-the-page)._
```

Comment:
```suggestion
**NB** : Some Bootswatch templates like lux or pulse won't work out of the box because of their tall navbar.
Add [CSS rules to prevent overlap with other elements](https://getbootstrap.com/docs/4.6/components/navbar/#placement) like [padding to the body](https://stackoverflow.com/questions/10336194/top-nav-bar-blocking-top-content-of-the-page).
```